### PR TITLE
Different Trigger for `OnPersist` and `PostPersist`

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo-modules/issues/354

### CODE REVIEW

File: src/neo/SmartContract/TriggerType.cs [hunk 1]
```diff
@@ -5,24 +5,25 @@ namespace Neo.SmartContract
...
         /// The application function can accept multiple parameters, change the states of the blockchain, and return any type of value.
         /// The contract can have any form of entry point, but we recommend that all contracts should have the following entry point:
         ///     public byte[] main(string operation, params object[] args)
         /// The functions can be invoked by creating an InvocationTransaction.
         /// </summary>
-        Application = 0x40,
+        Application = 1 << 4,
 
-        All = System | Verification | Application
+        All = SystemPersist | SystemPostPersist | Verification | Application
```
Comment (shargon) [inline src/neo/SmartContract/TriggerType.cs:27]: @erikzhang I think that we can remove flags from `TriggerType`, what do you think?

Comment (erikzhang) [reply src/neo/SmartContract/TriggerType.cs:27]: What do you mean?

Comment (shargon) [reply src/neo/SmartContract/TriggerType.cs:27]: We never use `TriggerType` as a flag

Comment (erikzhang) [reply src/neo/SmartContract/TriggerType.cs:27]: We have used flags to determine `AllowedTriggers`. Maybe it will be needed in the future?

**Review by Tommo-L — APPROVED** [review]

**Review by erikzhang — APPROVED** [review]

**Pull request merged**